### PR TITLE
Tiled Gallery: fix view rendering issues

### LINKED ISSUE #37194: Tiled Gallery block does not respect the `Circle` setting on live pages
### Impacted plugin

Jetpack

### Quick summary

Initially reported here: 8138345-zd-a8c

The user has set the `Tiled Gallery` block to display as `Circles` in the block styles as shown here: 

But when viewing the live page, they still show as squares/rectangles: 

I have not been able to find a workaround - tested on two of my own sites (Simple & AT) and could replicate on both, so the plan/themes don't seem to be a contributor.

### Steps to reproduce

1. Add a `Tiled Gallery` block to any page/post
2. Set the `Styles` for the block to `Circles`

### A clear and concise description of what you expected to happen.

The images should get cropped to be circles.

### What actually happened

The images are still showing as squares.

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple, Atomic

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes https://github.com/Automattic/jetpack/issues/37194

## Proposed changes:

This PR fixes the Tiled Gallery not being rendered as expected in the frontend. The issues was introduced by https://github.com/Automattic/jetpack/pull/36854.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Don't checkout this branch just yet
- Spin up a test site
- Connect Jetpack
- Create a new post
- Add a _Tiled Gallery_ block and select some images
- Checkout this branch
- Refresh the editor and verify there's no error message
- In the block styles panel, select _Circles_
Screenshot 2024-05-03 at 10 56 47 AM

- Open the preview
- Notice the items are circular
Screenshot 2024-05-03 at 10 54 27 AM

- Verify the other styles are rendered the same in the editor and frontend

### CODE REVIEW

**Review by coder-karen — COMMENTED** [review]
This works well in preview and for published pages. 
However I notice in testing locally and with a Jurassic Ninja site that in the editor the images stay as squares. Checking back a few Jetpack versions I can see the new styles (circles) were previously showing in the editor as well as on the front-end.

**Comment by jeherve** [comment]
> In the editor the images stay as squares.

Yes, that's what I'm seeing in my tests as well.

**Comment by monsieur-z** [comment]
The gallery was similar to the preview when I initially tested the PR, but testing again it's now different indeed! Not sure what I did there. Thanks for catching this.

**Comment by monsieur-z** [comment]
I've devised a new solution and updated the block `save` function to move the style-related class name one step higher. A new deprecated version was required for backward compatibility, so I've updated the instructions to take this into account.

File: projects/plugins/jetpack/extensions/blocks/tiled-gallery/view.scss [hunk 1]
```diff
@@ -6,12 +6,17 @@ $tiled-gallery-max-column-count: 20;
 $tiled-gallery-max-rounded-corners: 20; // See constants.js for JS counterpart
 
 .wp-block-jetpack-tiled-gallery {
-	&.is-style-circle .tiled-gallery__item img {
-		border-radius: 50%;
+	&.is-style-circle,
+	.is-style-circle {
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/tiled-gallery/view.scss:10]: Ensure the gallery is rendered correctly in exising posts.

**Review by coder-karen — APPROVED** [review]
Works well for me!

Testing with a previously added gallery, there were no errors when updating the branch either.

**Pull request merged**